pkg/types/spec/api.go

```diff
@@ -35,19 +35,18 @@ import (
 
 type API struct {
 	*userconfig.API
-	ID                    string                 `json:"id"`
-	SpecID                string                 `json:"spec_id"`
-	PredictorID           string                 `json:"predictor_id"`
-	DeploymentID          string                 `json:"deployment_id"`
-	Key                   string                 `json:"key"`
-	PredictorKey          string                 `json:"predictor_key"`
-	LastUpdated           int64                  `json:"last_updated"`
-	MetadataRoot          string                 `json:"metadata_root"`
-	ProjectID             string                 `json:"project_id"`
-	ProjectKey            string                 `json:"project_key"`
-	CuratedModelResources []CuratedModelResource `json:"curated_model_resources"`
-	LocalModelCaches      []*LocalModelCache     `json:"local_model_cache"` // local only
-	LocalProjectDir       string                 `json:"local_project_dir"`
+	ID               string             `json:"id"`
+	SpecID           string             `json:"spec_id"`
+	PredictorID      string             `json:"predictor_id"`
+	DeploymentID     string             `json:"deployment_id"`
+	Key              string             `json:"key"`
+	PredictorKey     string             `json:"predictor_key"`
+	LastUpdated      int64              `json:"last_updated"`
+	MetadataRoot     string             `json:"metadata_root"`
+	ProjectID        string             `json:"project_id"`
+	ProjectKey       string             `json:"project_key"`
+	LocalModelCaches []*LocalModelCache `json:"local_model_cache"` // local only
```

Comment:
@deliahu at a second look, it seems like removing `LocalModelCaches` is still necessary due to one reason: When re-deploying an API of whose model's contents have changed, without the `LocalModelCaches` present in the API spec, the API wouldn't get updated. The function responsible for this is `areAPIsEqual` in the `local/api.go` file.

Assuming the above situation doesn't exist, one small setback which I think falls outside the scope of this PR is that we'd need to start using the model IDs from the running containers to determine if a model can be removed from the Cortex cache or not - right now, we are relying on the API spec to tell us that. I think keeping this in the API spec is better because a container is ephemeral, whereas the API spec will be there after a reboot as well - just as the model will still be there.

Because of the first reason listed in this post, I think the presence of `LocalModelCaches` in the API spec is actually justified.